providers/linux/machineid.go

```diff
@@ -27,11 +27,37 @@ import (
 	"github.com/elastic/go-sysinfo/types"
 )
 
+var (
+	// Possible (current and historic) locations of the machine-id file.
+	// These will be searched in order.
+	machineIDFiles = []string{"/etc/machine-id", "/var/lib/dbus/machine-id", "/var/db/dbus/machine-id"}
+)
+
 func MachineID() (string, error) {
```

Comment:
What do you think about adding `machine-id` files to the testdata directory? Then make this unit testable by adding a `machineID(rootDir string)` that enables a test to point this at a specific directory in testdata. `MachineID()` would call `machineID("/"`).

Reply:
We certainly could. I assume it wouldn't do much more than `ioutil.ReadFile()` and `string(bytes.TrimSpace())` though, so I'm not sure how much testing it actually achieves.

Thanks for merging!